modules/runtime/nuxeo-runtime-migration/src/main/java/org/nuxeo/runtime/migration/MigrationService.java

```diff
@@ -234,4 +237,28 @@ public long getProgressTotal() {
      */
     void runStep(String id, String step);
 
+    /**
+     * @since 11.1
+     */
+    Migration getMigration(String id);
+
+    /**
+     * @since 11.1
+     */
+    List<Migration> getMigrations();
+
+    /**
+     * @throws IllegalStateException if the migration doesn't have exactly one available step to run.
+     * @since 11.1
+     */
+    void run(String migrationId) throws IllegalArgumentException;
+
+    /**
+     * Waits for completion of all migration steps.
+     *
+     * @return {@code true} if the migration executor has no more threads running {@code false} otherwise
+     * @since 11.1
+     */
+    boolean await(Duration duration) throws InterruptedException;
```

Comment:
Not sure we want this API, we don't have migration to wait during tests, and current migration tests don't need it.

Such APIs are a pain when they become available to the world. Here you may want it for tests, but later someone will ask you to make it work in a Nuxeo Cluster... Because it's public on an interface.

Reply:
I implemented it because in this new test I need to wait https://github.com/nuxeo/nuxeo/blob/c0c77d054898ec7a59bc35b08a50abf06822c8b5/modules/core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/migration/test/TestMigrationService.java#L72

First when I saw `nextTransaction()` was not helping I did `Thread.sleep(1000)` for the test to pass. Otherwise it will fail. Shall I get back to this solution ?

Reply:
If we don't want to expose it publicly, I would put that method only on the implementation class stating it's for test purposes, and use it in the test.
Does that sound good @kevinleturc ?

Reply:
MigrationFeature#Initialize calls the await method. If it is protected, then we can't call it

Reply:
Sounds good to have a protected method on the implementation only.

Not sure I, the end if we want the waiter, we control the migration start/stop, to discuss.

Reply:
if it is protected we can't call it in the MigrationFeature#initialize